src/coordinator.coffee

```diff
@@ -244,6 +249,17 @@ class Coordinator extends EventEmitter
         delete @process[k]
       return callback null, processes
 
+  updateParticipantMetadata: (node, metadata, callback) ->
+    metadata = {} unless metadata
+    process = null
+    for k, v of @processes
+      if k == node
+        process = v
+    unless process
+      return callback new Error "Node #{node} not defined"
+    @processes[k].metadata = metadata
```

Comment:
For `@connections` we already have the concept, as we add an entry immediately (with empty queue info).

However, both `@participants` and `@processes` have different shapes than the (desired) FBP graph. There may be multiple participants per role, it may not actually run. Processes is only for participants started by us, not those discovered externally.

So I'm thinking that we (not in this PR) should add a `@roles`, which would track the "nodes" (in FBP). Probably would make the `start/stopParticipant()` API internal, wrapped by a `add/removeRole()`